hnn_core/batch_simulate.py

```diff
@@ -76,20 +122,63 @@ def run(self, param_grid, return_output=True, combinations=True, n_jobs=1):
             Default is True.
         n_jobs : int, optional
             Number of parallel jobs. Default is 1.
+        backend : str or joblib.parallel.ParallelBackendBase instance, optional
+            The parallel backend to use. Can be one of `loky`, `threading`,
+            `multiprocessing`, or `dask`. Default is `loky`
+        verbose : int, optional
+            The verbosity level for parallel execution. Default is 50.
+        clear_cache : bool, optional
+            Whether to clear the results cache after saving each batch.
+            Default is False
 
         Returns
         -------
         results : list
             List of simulation results if return_output is True.
         """
+        _validate_type(param_grid, types=(dict,), item_name='param_grid')
+        _validate_type(n_jobs, types='int', item_name='n_jobs')
+        _check_option('backend', backend, ['loky', 'threading',
+                                           'multiprocessing', 'dask'])
+        _validate_type(verbose, types='int', item_name='verbose')
+
         param_combinations = self._generate_param_combinations(
             param_grid, combinations)
-        results = self.simulate_batch(param_combinations, n_jobs=n_jobs)
+        total_sims = len(param_combinations)
+        num_sims_per_batch = max(total_sims // self.batch_size, 1)
+        batch_size = min(self.batch_size, total_sims)
+
+        results = []
+        for i in range(batch_size):
+            start_idx = i * num_sims_per_batch
+            end_idx = start_idx + num_sims_per_batch
+            if i == batch_size - 1:
+                end_idx = len(param_combinations)
+            batch_results = self.simulate_batch(
+                param_combinations[start_idx:end_idx],
+                n_jobs=n_jobs,
+                backend=backend,
+                verbose=verbose)
+
+            if self.save_outputs:
+                self.save(batch_results, start_idx, end_idx)
+
+            if self.summary_func is not None:
+                summary_statistics = self.summary_func(batch_results)
```

Comment:
Do you think it's necessary to add the option to save summary statistics? Or should the save function be reserved for simulation results

Reply:
I think saving summary statistics can be a valuable feature for analysis. but, to maintain clarity and separation of concerns, it is better to handle the saving of summary statistics in a separate method rather than adding it to the existing `save` method. This ensures that the roles of saving detailed simulation results and derived summaries remain distinct. If you feel that this feature will add significant value, we can discuss it further in our meeting tomorrow. 

Additionally, I have some naming suggestions to improve clarity and consistency. The current parameter `file_path` could be renamed to `saveFolder`, following the NetPyNE simulation [documentation](http://doc.netpyne.org/user_documentation.html#simulation-configuration), to indicate the path where output data should be saved. Moreover, adding a parameter `filename` to specify the name of the file for saving model output (defaulting to 'sim_run_') would enhance usability.

Looking forward to discussing this further in our meeting.